src/xwayland.c

```diff
@@ -134,6 +121,29 @@ handle_destroy(struct wl_listener *listener, void *data)
 	view_destroy(view);
 }
 
+static void
+configure(struct view *view, struct wlr_box geo)
+{
+	assert(view->xwayland_surface);
+
+	view->pending_move_resize.x = geo.x;
+	view->pending_move_resize.y = geo.y;
+	view->pending_move_resize.width = geo.width;
+	view->pending_move_resize.height = geo.height;
+
+	wlr_xwayland_surface_configure(view->xwayland_surface, geo.x, geo.y,
+				       geo.width, geo.height);
+
+	/* If not resizing, process the move immediately */
+	if (view->w == geo.width && view->h == geo.height) {
+		view->x = geo.x;
+		view->y = geo.y;
+
+		wlr_scene_node_set_position(&view->scene_tree->node, geo.x, geo.y);
+		ssd_update_geometry(view);
```

Comment:
Do we need `ssd_update_geometry()` here?
If we're only moving, I think I'm right in saying that the ssd-nodes will move with the view-node.

Reply:
I expect you're right. Perhaps the `ssd_update_geometry()` in `view_move()` is also unnecessary?

Reply:
See 5d8eb02af6138a1a7d76229478f6aa1a204496d0

Reply:
Nice. It'd be good to just get @Consolatis eyes on the `ssd_update_geometry()` bit.

Reply:
We need to call `ssd_update_geometry()` for move as well because we adjust the invisible resize grab area on move in case they would be covering a layer-shell client. See
https://github.com/labwc/labwc/blob/06cc21664fd66a969012cb5da56a1bf32f1c0a1e/src/ssd/ssd.c#L189-L197

Reply:
Thanks. Well remembered :smile: I even looked at that code an didn't clock it!

Reply:
Okay that simplifies things, the post-processing for move and move+resize is the same then. Thanks for checking!